File: java/niven.py
# sample using pythonv3.8
num = 18
n1 = num % 10
n2 = num // 10 % 10
print(n1, n2)
# sum of all digits is equal to product of all digits
# sample 1124 1x1x2x4 = 8 and 1+1+2+4 = 8
def getspy(number):
    n1 = number % 10
    n2 = number // 10 % 10
    n3 = number // 100 % 10
    n4 = number // 1000 % 10
    sum = n1 + n2 + n3 + n4
    multiply = n1 * n2 * n3 * n4
    if sum == multiply:
        print('number is spy number')
    else:
        print('number is not spy number')
    # print( n1, n2, n3, n4 )
input_number = int(input('enter a number: '))
getspy(input_number)

File: java/niven.java
package java;
import java.util.Scanner;
public class niven {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the value: ");
        int num = sc.nextInt();
        // sample input 18 so 1+8 = 9 and 18 is divisible by 9
        int n1 = num%10; 
        int n2 = num / 10 % 10;
        int sum = n1+n2;
        if(num%sum==0){
            System.out.println("The number is niven number");
        }
        else{
            System.out.println("The number is not niven number");
        }
    }
}
